### PR TITLE
Serving cached AOT methods at the JITServer and loading them at the client

### PR DESCRIPTION
**Pull request by AlexeyKhrabrov**
This PR completes the minimal implementation of caching compiled code at JITServer. It adds the code to serve serialized methods from the JITServer AOT cache and load them at the client.

### CODE REVIEW

**Comment by AlexeyKhrabrov** [comment]
@mpirvu This PR is ready for review.

**Comment by AlexeyKhrabrov** [comment]
Added AOT cache statistics.

**Review by mpirvu — COMMENTED** [review]
LGTM. Please see the inline comments/questions.

**Comment by mpirvu** [comment]
jenkins compile all jdk8

File: runtime/compiler/control/J9Options.cpp [hunk 1]
```diff
@@ -2113,6 +2118,13 @@ bool J9::Options::preProcessJitServer(J9JavaVM *vm, J9JITConfig *jitConfig)
          jitConfig->clientUID = 0;
          jitConfig->serverUID = 0;
          }
+
+      if ((compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER) &&
+          compInfo->getPersistentInfo()->getJITServerUseAOTCache())
```
[the remark this answers is not in the document]
Comment (AlexeyKhrabrov) [reply runtime/compiler/control/J9Options.cpp:2123]: Yes. I though I had that test here. Will fix.

File: runtime/compiler/control/JITClientCompilationThread.cpp [hunk 2]
```diff
@@ -2899,14 +2885,11 @@ remoteCompilationEnd(
...
-               TR_Vlog_JITServer,
-               "JITClient: Applying remote AOT relocations to newly AOT compiled body for %s @ %s",
-               comp->signature(),
-               comp->getHotnessName()
-               );
-            }
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+               "JITClient: Applying remote AOT relocations to %s AOT body for %s @ %s",
+               comp->isDeserializedAOTMethod() ? "deserialized" : "newly compiled",
+               comp->signature(), comp->getHotnessName()
```
[the remark this answers is not in the document]
Comment (AlexeyKhrabrov) [reply runtime/compiler/control/JITClientCompilationThread.cpp:2895]: A freshly compiled method is not sent to the client in serialized form.

**Comment by AlexeyKhrabrov** [comment]
I've addressed the comments.

The 32-bit Windows build in the tests https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/1255/ is failing with an error unrelated to this PR: https://openj9-jenkins.osuosl.org/job/Build_JDK8_x86-32_windows_Personal/132/

```
21:36:51  F:\Users\jenkins\workspace\Build_JDK8_x86-32_windows_Personal\openj9\runtime\compiler\runtime\SignalHandler.c(401) : error C2220: warning treated as error - no 'object' file generated
21:36:51  F:\Users\jenkins\workspace\Build_JDK8_x86-32_windows_Personal\openj9\runtime\compiler\runtime\SignalHandler.c(401) : warning C4133: 'function' : incompatible types - from 'J9JavaVM *' to 'J9VMThread *'
21:36:51  make[4]: *** [runtime/compiler/CMakeFiles/j9jit.dir/build.make:4086: runtime/compiler/CMakeFiles/j9jit.dir/runtime/SignalHandler.c.obj] Error 2
```

The aix build seems to be stuck.

**Review by mpirvu — APPROVED** [review]

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit zlinuxjit  jdk11

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit,zlinuxjit jdk11

**Comment by pshipton** [comment]
I reverted the problematic PR causing the win32 build failure last night.

**Comment by mpirvu** [comment]
https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/1257/
Tests on zlinux passed.
Tests on xlinux failed due to infra as far as I can tell:
```
[Pipeline] nodesByLabel
05:20:23  Found a total of 20 nodes with the 'ci.role.test&&hw.arch.x86&&sw.os.linux' label
[Pipeline] echo
05:20:23  machine limit is 20
[Pipeline] timeout
05:20:23  Timeout set to expire in 1 hr 0 min
[Pipeline] {
[Pipeline] copyArtifacts
[Pipeline] }
[Pipeline] // timeout
[Pipeline] echo
05:20:54  Cannot get cached TRSS JSON data. Skipping copyArtifacts...
[Pipeline] timeout
05:20:54  Timeout set to expire in 2 hr 0 min
[Pipeline] {
[Pipeline] copyArtifacts
[Pipeline] }
[Pipeline] // timeout
[Pipeline] echo
07:06:02  Cannot run copyArtifacts from test.getDependency. Skipping copyArtifacts...
[Pipeline] sh
07:06:02  cent7-x64-5 was marked offline: Timed out for last 5 attempts
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
terminateJavaProcesses iteration: 1
[Pipeline] echo
PROCESSCATCH: Checking for any leftover java processes on the machine
[Pipeline] sh
cent7-x64-5 was marked offline: Timed out for last 5 attempts
[Pipeline] }
[Pipeline] // timeout
[Pipeline] echo
Exception: java.io.IOException: Unable to create live FilePath for cent7-x64-5
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
Finished: FAILURE
```

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
Even though the the test on xlinux appears as aborted; https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/1259/
all the tests were successful:
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_0/47/tapResults/
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_1/48/tapResults/

**Pull request merged**